### PR TITLE
Render Replica npm-style Badge in doc-template

### PR DESCRIPTION
**Pull request by amichaelparker**
### Summary
Ref: [This comment](https://github.com/cerner/terra-core/issues/1813#issuecomment-518840423). This renders a replica npm badge to replace the dynamically generated one from badgen. This allows projects that don't have their packages hosted on npm to not have a 404 badge on their doc site.

Closes [#1813](https://github.com/cerner/terra-core/issues/1813).

### Additional Details
This adds packageVersion and packageUrl props, and a badge will be rendered based on what is provided. If neither are, it will fall back to the current badgen badge. It also addresses an issue with Voiceover being able to jump out of focus trapped modals to the badge in some circumstances.

### CODE REVIEW

File: packages/terra-doc-template/src/DocTemplate.jsx [hunk 1]
```diff
@@ -92,11 +102,24 @@ const DocTemplate = ({
...
   }
 
-  const badge = (
-    <a href={`https://www.npmjs.org/package/${packageName}`}>
-      <img src={`https://badgen.net/npm/v/${packageName}`} alt="NPM version" />
-    </a>
-  );
+  const badge = packageVersion
+    ? (
+      <div className={cx('badge-container')}>
```
Comment (bjankord) [inline packages/terra-doc-template/src/DocTemplate.jsx]: Let's turn this into a component within the terra-doc-template package. It should make testing easier this way. I'm thinking a new file, `packages/terra-doc-template/src/NpmBadge.jsx` that we can import here.

Comment (amichaelparker) [reply packages/terra-doc-template/src/DocTemplate.jsx]: Changed! [af52c67](https://github.com/cerner/terra-core/pull/2571/commits/af52c67081f807649628e8e8f07b4e9957635768)

File: packages/terra-doc-template/src/NpmBadge.module.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
+:local {
+  .badge {
+    color: #fff;
+    font-family: Verdana,DejaVu Sans,sans-serif;
+    font-size: 11px;
```
Comment (bjankord) [inline packages/terra-doc-template/src/NpmBadge.module.scss]: let's update this to `0.7857rem`

File: packages/terra-doc-template/src/NpmBadge.module.scss [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    text-decoration: none;
+  }
+
+  .badge-container {
+    margin-bottom: 0.4em;
+  }
+
+  .badge-name,
+  .badge-version {
+    padding: 3.25px 5px;
```
Comment (bjankord) [inline packages/terra-doc-template/src/NpmBadge.module.scss]: let's update this to `0.232rem 0.357rem`

File: packages/terra-doc-template/src/terra-dev-site/doc/example/BadgeExamples.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+// eslint-disable-next-line import/no-extraneous-dependencies, import/no-unresolved, import/extensions
+import NpmBadge from '../../../NpmBadge';
+import { name } from '../../../../package.json';
+
+const packageUrl = 'https://engineering.cerner.com/terra-ui/components/terra-doc-template/doc-template/doc-template';
+const packageVersion = '2.13.0';
```
Comment (bjankord) [inline packages/terra-doc-template/src/terra-dev-site/doc/example/BadgeExamples.jsx]: For this file, let's actually pull the version from the package.json file.

```jsx
import { name, version } from '../../../../package.json';
```
and then pass `version` to the `packageVersion` prop. For the tests files, we can keep the version static.

File: packages/terra-doc-template/src/NpmBadge.module.scss [hunk 5]
```diff
@@ -0,0 +1,28 @@
+:local {
+  .badge {
+    color: #fff;
+    font-family: Verdana,DejaVu Sans,sans-serif;
+    font-size: 0.7857rem;
```
Comment (emilyrohrbough) [inline packages/terra-doc-template/src/NpmBadge.module.scss]: Should we have been overriding font-size and font-family ?

File: packages/terra-doc-template/src/DocTemplate.jsx [hunk 6]
```diff
@@ -17,6 +18,14 @@ const propTypes = {
    * The given component's npm package name.
    */
   packageName: PropTypes.string,
+  /**
+   * The url for the provided package if not in npm. This will be ignored if packageVersion is not set.
+   */
+  packageUrl: PropTypes.string,
```
Comment (emilyrohrbough) [inline packages/terra-doc-template/src/DocTemplate.jsx]: What url do we expect people to provide internally? Will this differ from the component source code URL we already supply on the page? 

What if someone want so to render the badge with a version but doesn't care to supply a link?

**Comment by emilyrohrbough** [comment]
>  It also addresses an issue with Voiceover being able to jump out of focus trapped modals to the badge in some circumstances.

Was voice over functionally verified for these changes?

File: packages/terra-doc-template/src/NpmBadge.module.scss (hunk 5 above)
Comment (amichaelparker) [reply packages/terra-doc-template/src/NpmBadge.module.scss]: The normal npm badge also uses its own font size and family. This allows the non-badgen version of the badge to maintain consistency with the dynamic version.

File: packages/terra-doc-template/src/DocTemplate.jsx (hunk 6 above)
Comment (amichaelparker) [reply packages/terra-doc-template/src/DocTemplate.jsx]: They might want to point it to an internal source, or they may just want it to loop right back to the same page the badge is found on to simply not have a 404 badge.

If they supply a version without a url, it falls back to passing the version to the normal npm link, pointed at that version of the package. If that doesn't exist, the url prop gives them the option to point it wherever they desire.

**Pull request merged**